### DISCUSSION TITLE
remote management reports "read: 3 Connection reset by peer"

### DISCUSSION THREAD

**Discussion by fluca1978** [pgagroal/pgagroal#269] [General]
I'm trying to figure out how to use `pgagroal-cli` remotely, so I could have mess something.
However, running version 7918ce41364b45d5700a3b7fe323bde0236cffb9 and doing this:
- start `pgagroal`
- remove the `/etc/pgagroal/pgagroal.conf` configuration file
- invoke `pgagroal-cli` on localhost:
```
$ pgagroal-cli  -h localhost -p 5432 -U luca details
Password : 
WARN  management.c:553 pgagroal_management_read_status: read: 3 Connection reset by peer
WARN  management.c:695 pgagroal_management_read_details: read: 3 Invalid argument

$ pgagroal-cli  -h localhost -p 5432 -U luca is-alive
Password : 
WARN  management.c:947 pgagroal_management_read_isalive: read: 3 Connection reset by peer
```

On the log of `pgagroal` (running at `debug` level):

```
DEBUG pool.c:1193 pgagroal_pool_status: State: IN_USE
DEBUG pool.c:1194                       Slot: 3
DEBUG pool.c:1195                       Server: 0
DEBUG pool.c:1196                       User: luca
DEBUG pool.c:1197                       Database: admin
DEBUG pool.c:1198                       AppName: pgagroal
DEBUG pool.c:1199                       Rule: -1
DEBUG pool.c:1200                       Time: Wed Jun 15 06:18:45 2022
DEBUG pool.c:1201                       FD: 6
(libev) epoll_wait: Bad file descriptor
```

I'm not understanding how to use remote management, since it appear an `admin` database.

**Comment by jesperpedersen** [comment]
Moved to "General".

Remote management is for access from an internal network, where you don't have access to the machine/VM/... itself.

You should not delete the configuration, and the "admin" database is a "virtual" one.

I havn't ported the tutorials yet, but you can look at https://github.com/pgmoneta/pgmoneta/blob/main/doc/tutorial/03_remote_management.md - it will be similar.

**Comment by fluca1978** [comment]
It seems to me I'm doing either something wrong or `pgagroal-cli` is not taking into account the `host` and `port` options.
Having cofnigured `management = 6432`, and running `pgagroal`, the following seems to do the trick:

```
$ pgagroal-cli -h localhost -p 6432 status
Status:              Running
Active connections:  0
Total connections:   1
Max connections:     4
```

but it is not using the networkng layer, and in fact, if I use the wrong port:

```
$ pgagroal-cli -h localhost -p 2345 status
Status:              Running
Active connections:  0
Total connections:   1
Max connections:     4
```

it works again!
The problem seems to be here <https://github.com/agroal/pgagroal/blob/master/src/cli.c#L416>: since the program finds the configuration file (and correctly loads it), it does not honor remote management parameters.

If I remove the configuration file, then it asks for a remote user and a password:

```
# while pgagroal is running..
$ mv /etc/pgagroal/pgagroal.conf /etc/pgagroal/pgagroal.conf.old

$ pgagroal-cli -h localhost -p 6432 status
User name: luca
Password : 
Status:              Running
Active connections:  0
Total connections:   1
Max connections:     4
```

So i went thru a couple of potential issues:
1) even if `management` is configured, on localhost the `pgagroal-cli` does not honor the `host` and `port` options, preferring the local connection as first. This in my opinion is a bug.
2) I named the file badly, so instead of `pgagroal_admins.conf` I named if `pgagroal_admin_users.conf`. In such case, the remote management was telling me `bad credentials`. I think the system should report back that there are no users configured at all.
3) last, we need to document a little more this usage.

**Reply by jesperpedersen** [reply]
1. Yes, we need `configuration_path != NULL && host == NULL && port == NULL` in L416 - plus some other checks likely
2. You can use any file name with `-A` - default will always be `/etc/pgagroal/pgagroal_admins.conf`. There is a `WARN` if `management` is set, and there are no admin users. We should not reply with unknown user as it could lead to attacks.
3. Yes, I'll add a tutorial on the subject
